plugins/dynamic-few-shot-lambda/README.md

```diff
@@ -0,0 +1,454 @@
+        S[serialized_document: Document metadata]
+    end
+
+    subgraph "Output Payload"
+        T[system_prompt: Unchanged]
+        U[task_prompt_content: Array with Prompt segments and Example images]
+    end
+
+    D -.-> P
+    D -.-> Q
+    D -.-> R
+    D -.-> S
+
+    N -.-> T
+    N -.-> U
+
+    subgraph "S3 Vectors Infrastructure"
+        X[Vector Bucket: Encrypted storage]
+        Y[Vector Index: 3072-dim cosine similarity]
+        Z[Metadata: classPrompt, attributesPrompt, imagePath]
+    end
+
+    H -.-> X
+    H -.-> Y
+    H -.-> Z
+```
+
+## Quick Start
+
+### Step 1: Deploy the Dynamic-few shot Stack
+
+```bash
+# Navigate to the dynamic-few-shot-lambda directory
+cd plugins/dynamic-few-shot-lambda
+
+# Deploy using AWS SAM
+sam deploy --guided
+```
+
+### Step 2: Get the Lambda ARN
+
+After deployment, get the ARN from CloudFormation outputs:
+
+```bash
+aws cloudformation describe-stacks \
+    --stack-name GENAIIDP-dynamic-few-shot-stack \
+    --query 'Stacks[0].Outputs[?OutputKey==`DynamicFewShotFunctionArn`].OutputValue' \
+    --output text
+```
+
+### Step 3: Populate the Examples Dataset
+
+Use the [fewshot_dataset_import.ipynb](notebooks/fewshot_dataset_import.ipynb) notebook to import a dataset into S3 Vectors, or manually upload your example documents and metadata to the S3 bucket and vector index created by the stack.
+
+### Step 4: Configure IDP to Use Dynamic Few-Shot
+
+Add the Lambda ARN to your IDP extraction configuration:
+
+```yaml
+extraction:
+  custom_prompt_lambda_arn: "arn:aws:lambda:region:account:function:GENAIIDP-dynamic-few-shot"
+```
+
+**Important**: Your extraction task prompt must include the `{FEW_SHOT_EXAMPLES}` placeholder where you want the dynamic examples to be inserted.
+
+### Step 5: Run the Demo Notebook
+
+0. Run `notebooks/examples` steps 0, 1, 2
+1. Open `plugins/dynamic-few-shot-lambda/notebooks/step3_extraction_with_custom_lambda.ipynb`
+2. Run all cells to see the comparison
+
+## Lambda Interface
+
+### Input Payload Structure
+
+The Lambda receives the full IDP context as a custom prompt Lambda:
+
+```json
+{
+  "config": {
+    "extraction": {...},
+    "classes": [...],
+    ...
+  },
+  "prompt_placeholders": {
+    "DOCUMENT_TEXT": "Full OCR text from all pages",
+    "DOCUMENT_CLASS": "invoice", 
+    "ATTRIBUTE_NAMES_AND_DESCRIPTIONS": "LineItems: List of line items in the invoice...",
+    "DOCUMENT_IMAGE": ["s3://bucket/document/page1.jpg", "s3://bucket/document/page2.jpg"]
+  },
+  "default_task_prompt_content": [
+    {"text": "Resolved default task prompt..."},
+    {"image_uri": "s3://..."}, // if images present
+    {"cachePoint": true} // if cache points present
+  ],
+  "serialized_document": {
+    "id": "document-123",
+    "input_bucket": "my-bucket",
+    "pages": {...},
+    "sections": [...],
+    ...
+  }
+}
+```
+
+### Output Payload Structure
+
+The Lambda returns modified prompt content with dynamic few-shot examples:
```

Comment:
Can you make your demo notebook display the images / text of the chosen examples, and compare to the image / text for the document being processed?  
It will be powerful, and helpfully, to be able to visually anchor the selected examples.. Do this for two or three visually different invoices so it's easy to see that the embeddings are selecting similar examples.
You can use this in your demo as well to highlight how the example selection works.. visually.
Include screenshots of these examples (original docs + selected examples for each) in this README, for impact.